unittest/unittest.py

```diff
@@ -214,11 +232,15 @@ def test_cases(m):
             if isinstance(c, object) and isinstance(c, type) and issubclass(c, TestCase):
                 yield c
 
-    m = __import__(module)
+    m = __import__(module) if isinstance(module, str) else module
```

Comment:
I smell this is not related to "log failure tracebacks at the end". Ok for this time, but next time I'll ask to make this a separate commit ;-).

Reply:
I forgot about this change sneaking in as well. I will certainly split it into a separate commit at least.